### ISSUE TITLE
Connect-DbaInstance's ReadOnly field doesn't describe RO AG databases

### ISSUE THREAD

**Issue by LowlyDBA** [sqlcollaborative/dbatools#3691]
### Before submitting a bug report:

- [x] Ensure you are able to reproduce it on the latest released version (_we release often_)
- [x] Verified this bug is not already reported in an issue
- [x] Verified errors are not related to permissions
- [x] Can reproduce in a clean PowerShell session (_clean = `powershell -NoProfile`_)

## Steps to Reproduce
Run the following on a server which contains a read-only replica database in an AG:
```powershell
(Connect-DbaInstance -SqlInstance xxx).Databases
```

## Expected Behavior
Database status should return as ReadOnly = True. If `DATABASEPROPERTYEX('DatabaseName', 'Updateability');` is used to get this value instead, it provides a read only status that takes into account its AG role.
```
ReadOnly                                    : True
```

## Actual Behavior

```
ReadOnly                                    : False
```

## Environmental data

```
Name                           Value
----                           -----
PSVersion                      5.1.14393.2248
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.14393.2248
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

## Impact

This is causing dbachecks' Last Full Backup check to incorrectly flag secondary nodes as not having had a full backup.

**Comment by wsmelton** [comment]
I don't follow the expected behavior you are showing... Unless mobile view is leaving something off.

**Comment by wsmelton** [comment]
The status is an SMO property, it is not something we manipulate not change... It returns what it returns.

**Comment by LowlyDBA** [comment]
@wsmelton  Yeah, I'm suggesting to use a query for that value since SMO doesn't return the proper read only status when a DB is read only as a secondary AG replica.

**Comment by wsmelton** [comment]
And what state is returned?

What is returned by Get-DbaDatabase?

I see no reason to modify it to a query. Just validate if it is an AG db.

**Comment by LowlyDBA** [comment]
I adjusted the white space in the issue if that was the problemyou're having....it currently returns "False" for the ReadOnly attribute, which is not accurate. The output of this module makes it clear it is part of an AG but not specify if it is primary/secondary nor properly output its ReadOnly status. I'm confused why you're against adjusting it so it doesn't return the wrong ReadOnly status in this case, even if it means eschewing SMO for this data point. 

`Get-DbaDatabase` returns "False" for IsAccessible for a read-only secondary AG.

**Comment by wsmelton** [comment]
Because the database is not set to read-only, that is what that property is
meant for. Just like if it was a mirroring secondary.

If you set a database to read-only (the formal state) then that will return
true.

What you are referring to is out of scope for the database and does not
apply in an AG configuration.
